Describe web Gantt (#106)
DevExpress/dotnet-eud: interface-elements-for-web/articles/gantt/data-editing/modify-tasks.md

---
title: Add, Edit and Delete Tasks
author: Anna Kondratova
---
# Add, Edit and Delete Tasks

## Add a Task

> Right-click a task and select the "Add" [context menu item](~/interface-elements-for-web/articles/gantt/context-menu.md) or click the "Add Task" [toolbar item](~/interface-elements-for-web/articles/gantt/toolbar.md) to create a task:

Reviewer: ```suggestion
To create a task, right-click an existing task and select the "Add" [context menu item](~/interface-elements-for-web/articles/gantt/context-menu.md), or click the "Add Task" [toolbar item](~/interface-elements-for-web/articles/gantt/toolbar.md):
```

> - "New Task" - Appends a new task at the same level as the selected task.

Reviewer: ```suggestion
- "New Task" - Adds a new task at the same level as the selected task.
```

    ![](~/interface-elements-for-web/images/Gantt/add-new-task.png)

- "New Subtask" - Adds a child task to the selected task.

    ![](~/interface-elements-for-web/images/Gantt/add-new-subtask.png)

## Edit a Task

Use one of the approaches described below to edit a task.

* Double-click a task in the task list. 

    ![](~/interface-elements-for-web/images/Gantt/inline-editing.png)

> *  Right-click a task and select the "Task Details" context menu item (or double-click a task in the chart area).

Reviewer: ```suggestion
* Right-click a task and select the "Task Details" context menu item (or double-click a task in the chart area).
```

    ![](~/interface-elements-for-web/images/Gantt/popup-edit-form.png)

> * Hover the mouse pointer over the task you want to modify.

Reviewer: ```suggestion
* Move the mouse pointer over the task you want to modify.
```

    ![](~/interface-elements-for-web/images/Gantt/drag-and-drop.png)

>     **Move** - Hold on the task and drag it to change its start and end time.

Reviewer: I'm not sure what you mean by "Hold on the task". Do you mean:

"Drag the edge of a task to change its start and end time."

If not, please clarify.

Author: What you describe is resizing - change start **or** end time.

Moving is changing both start **and** end times.

Hold is to move the mouse pointer over the task and press on it.

![image](https://user-images.githubusercontent.com/27012218/82540850-35c8c980-9b58-11ea-9caa-fa9cce7a8be1.png)

Reviewer: Thank you for the detailed explanation. Can we say something like:

**Move** - Drag the task's left boundary to change its start time. Drag the right boundary to change its end time.

Author: The first sentence is better for me, thank you. Applied.

    ![](~/interface-elements-for-web/images/Gantt/moving.png)

>     **Resize** - Drag the resize handles at the task's edges to change the task duration.

Reviewer: ```suggestion
    **Resize** - Drag a resize handle (located at a task's edge) to change the task duration.
```

    ![](~/interface-elements-for-web/images/Gantt/resizing.png)

    **Change progress** - Move the slider to change the task's progress.

    ![](~/interface-elements-for-web/images/Gantt/change-progress.png)

## Save Changes

> The Gantt saves the changes right after you have edited a task in the task list or used drag and drop functionality.

Reviewer: ```suggestion
The Gantt saves changes right after you edit a task in the task list or perform a drag-and-drop operation.
```

![](~/interface-elements-for-web/images/Gantt/manage-tasks.gif)

> In the "Task Details" dialog, click OK to save the changes. 

Reviewer: ```suggestion
In the "Task Details" dialog, click OK to save changes. 
```

![](~/interface-elements-for-web/images/Gantt/dialog-ok.png)

## Remove a Task

> Right-click a task and select the 'Delete Task' [context menu](~/interface-elements-for-web/articles/gantt/context-menu.md) or [toolbar](~/interface-elements-for-web/articles/gantt/toolbar.md) item to remove a task.

Reviewer: ```suggestion
To remove a task, right-click a task and select the 'Delete Task' [context menu](~/interface-elements-for-web/articles/gantt/context-menu.md) item or [toolbar](~/interface-elements-for-web/articles/gantt/toolbar.md) item.
```

![](~/interface-elements-for-web/images/Gantt/delete-task.png)

## Undo/Redo Changes

> Use Ctrl+Z and Ctrl+Y or "Undo" and "Redo" [toolbar items](~/interface-elements-for-web/articles/gantt/toolbar.md) to roll back changes.

Reviewer: ```suggestion
Press Ctrl+Z or Ctrl+Y (or click "Undo" or "Redo" in the [toolbar](~/interface-elements-for-web/articles/gantt/toolbar.md)) to cancel or reverse changes.
```
